maestro-client/src/main/java/maestro/Maestro.kt

```diff
@@ -536,11 +507,67 @@ class Maestro(private val driver: Driver) : AutoCloseable {
         return driver.isUnicodeInputSupported()
     }
 
+    private fun waitUntilScreenIsStatic(timeoutMs: Long): Boolean {
+        if (isIOS) {
+            return MaestroTimer.retryUntilTrue(timeoutMs) {
+                val screenChanged = driver.isScreenChanged()
+
+                LOGGER.info("screen changed = $screenChanged")
+                return@retryUntilTrue !screenChanged
+            }
+        } else {
+            return MaestroTimer.retryUntilTrue(timeoutMs) {
+                val startScreenshot: BufferedImage? = tryTakingScreenshot()
+                val endScreenshot: BufferedImage? = tryTakingScreenshot()
+
+                if (startScreenshot != null &&
+                    endScreenshot != null &&
+                    startScreenshot.width == endScreenshot.width &&
+                    startScreenshot.height == endScreenshot.height
+                ) {
+                    val imageDiff = ImageComparison(
+                        startScreenshot,
+                        endScreenshot
+                    ).compareImages().differencePercent
+
+                    return@retryUntilTrue imageDiff <= SCREENSHOT_DIFF_THRESHOLD
+                }
+
+                return@retryUntilTrue false
+            }
+        }
+    }
+    private fun screenshotWaitForAppToSettle(initialHierarchy: ViewHierarchy?): ViewHierarchy {
+        val timeout = ANIMATION_TIMEOUT_MS
+        LOGGER.info("Waiting for animation to end with timeout $timeout")
+        val didFinishOnTime = waitUntilScreenIsStatic(SCREEN_SETTLE_TIMEOUT_MS)
+
+        return if (didFinishOnTime) viewHierarchy() else genericWaitForAppToSettle(initialHierarchy)
+    }
+    private fun genericWaitForAppToSettle(initialHierarchy: ViewHierarchy?): ViewHierarchy {
```

Comment:
Can we create one method for waitForAppSettle and pass comparing method as an input. Maybe we can have enum Hierarchy and Screenshot being two members